### ISSUE TITLE
Template validation as part of pull requests

### ISSUE THREAD

**Issue by jouvin** [quattor/template-library-core#38]
As discussed during the workshop, it'd be nice to have in all template-library-xxx repositories a validation test associated with pull requests. The initial implementation of the test could be the execution of create-vanilla-SCDB.sh (in SCDB repo). This script pulls down everything needed to compile the examples and compile them. We may just have to check that the script properly report a failure status.

**Comment by stdweird** [comment]
@jouvin i'm testing latest scdb master, i get
```
jenkins@jenkins1:~/workspace/test_scdb_build_all/scdb/utils/scdb$ ./create-vanilla-SCDB.sh -F HEAD
Creating vanilla SCDB from ./../.. in /tmp/scdb-vanilla...
Adding ant version apache-ant-1.9.4...
Adding panc version panc-10.1...
Adding scdb-ant-utils version scdb-ant-utils-9.0.2...
Adding svnkit version svnkit-1.8.6...
Downloading template libary (version: HEAD)...
/tmp/quattor-template-library/release/src/scripts/get-template-library: 256: /tmp/quattor-template-library/release/src/scripts/get-template-library: Bad substitution
Failed to download template libary. Check your options: must be valid for this script or get-template-library
usage:  create-vanilla-SCDB.sh [-F] [--debug] [-d scdb_dir] [options...] [quattor_version]

    Valid options are the following ones plus all valid get-template-library options:

        -d scdb_dir : directory where to create SCDB.
                      (D: /tmp/scdb-vanilla)
        --debug : debug mode. Checkout rather than export templates
        -F : remove scdb_dir if it already exists.
```

it would also be nice if the script would use
```
git_clone_root=${TMPDIR:-/tmp}/quattor-template-library
scdb_dir=${TMPDIR:-/tmp}/scdb-vanilla
```
so i can change the /tmp location on jenkins

**Comment by jouvin** [comment]
The error you report is weird... Are you sure that you use the last version of `create-vanilla-SCDB.sh` and that you don't have a version of `get-template-library` located in the same directory as `create-vanilla-SCDB.sh` (this will prevent retrieving the last version from GitHub).

BTW, I spent some time today fixing `--pull-request` and it should now really work as documented... in the help of `get-template-library` or in the upcoming new version of http://www.quattor.org/documentation/2014/06/06/how-to-use-template-library.html (see https://github.com/quattor/quattor.github.com/pull/131 in the meantime).

I'm adding the modification you asked for to be able to relocate `/tmp`.

**Comment by stdweird** [comment]
@jouvin it was a clone in a new directory. i'll look into it again this week

**Comment by jouvin** [comment]
@stdweird Looking again at the error (I missed the long line...), I think this is caused by a bug I fixed during the last day of the workshop, where the shell was `sh` which doesn't necessarily translate to `bash`... See https://github.com/quattor/scdb/pull/32.

**Comment by stdweird** [comment]
@jouvin ah yes, you're right

```
jenkins@jenkins1:~$ ls -l /bin/sh
lrwxrwxrwx 1 root root 4 Nov 13  2013 /bin/sh -> dash
```

**Comment by jouvin** [comment]
Download the last version of `create-vanilla-SCDB`.

**Comment by jouvin** [comment]
@stdweird let me know if there is anything I could do to help getting this working...

**Comment by jouvin** [comment]
It'd be good to have it working before 15.4...

**Comment by jouvin** [comment]
@stdweird what can we do to complete this...??? Issues found during RC2 would have been avoided if running the suggested unit tests... I'm reassigning to 15.6 with the hope that we can get it working before that.

**Comment by jrha** [comment]
I'm pretty sure this was actually implemented in #206.
